src/multi_physics/QED/CMakeLists.txt

```diff
@@ -1,19 +1,44 @@
-cmake_minimum_required(VERSION 3.5)
+add_library(QED INTERFACE)
+add_library(${PROJECT_NAME}::QED ALIAS QED)
 
-set(CMAKE_CXX_EXTENSIONS OFF)
+target_compile_features(QED INTERFACE cxx_std_14)
 
-###Header only library called "pxrmp_qed"
-add_library(pxrmp_qed INTERFACE)
+if(PXRMP_QED_OMP)
+   find_package(OpenMP REQUIRED)
+   target_link_libraries(QED INTERFACE OpenMP::OpenMP_CXX)
+   target_compile_definitions(QED INTERFACE PXRMP_HAS_OPENMP=1)
+endif()
 
-###Finds Boost library
-find_package(Boost REQUIRED)
-include_directories(${Boost_INCLUDE_DIRS})
+find_package(Boost 1.66.0)
+if(Boost_FOUND)
+    target_link_libraries(QED INTERFACE Boost::boost)
+    target_compile_definitions(QED INTERFACE PXRMP_HAS_BOOST=1)
+endif()
 
-target_include_directories(pxrmp_qed INTERFACE src/)
-target_include_directories(pxrmp_qed INTERFACE src/containers)
-target_include_directories(pxrmp_qed INTERFACE src/math)
-target_include_directories(pxrmp_qed INTERFACE src/utils)
-target_include_directories(pxrmp_qed INTERFACE src/physics)
-target_include_directories(pxrmp_qed INTERFACE src/physics/schwinger)
-target_include_directories(pxrmp_qed INTERFACE src/physics/breit_wheeler)
-target_include_directories(pxrmp_qed INTERFACE src/physics/quantum_sync)
+if(PXRMP_QED_TABLEGEN)
+    if(NOT Boost_FOUND)
+         message( FATAL_ERROR "Table generation requires Boost (>= 1.66)!" )
+    endif()
+endif()
+
+target_include_directories(QED INTERFACE src/)
```

Comment:
I would potentially skipp all but the `src/` include and add a prefix `picsar/` generally inside it to avoid collisions with third party libraries.